From a87e0358b8adfd19808c0007d3c32c00fc51b3d8 Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert+renesas@glider.be>
Date: Fri, 8 May 2015 16:23:36 +0200
Subject: [PATCH 11/12] kbuild: Create a rule for building device tree overlay
 objects

Modify "cmd_dtc" to use the "dtb" output format for ".dtbo" files.
Add a rule to build ".dtbo" files from ".dtso" files.

This allows to handle device tree overlays like plain device trees.

Signed-off-by: Geert Uytterhoeven <geert+renesas@glider.be>
---
v4:
  - Rebase on top of commit dd7699e37f289fa4 ("Revert "kbuild: Create
    directory for target DTB""),

v3:
  - Commit 78046fabe6e7807a ("kbuild: determine the output format of DTC
    by the target suffix") dropped the second parameter of cmd_dtc
    again:
      - Drop the second parameter,
      - Treat output format "dtbo" as "dtb".
  - Add dtc dependency, cfr. commit b8fc5b2157b10dad ("kbuild: add dtc
    as dependency on .dtb files"),
  - Move rule up, just after the .dtb rule,

v2:
  - Since commit 4f0e3a57d6eb727c ("kbuild: Add support for DT binding
    schema checks"), cmd_dtc takes a second parameter specifying the
    output format.

squash! kbuild: Create a rule for building device tree overlay objects
---
 scripts/Makefile.lib | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index aed0dfff505e..28b8f706650a 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -321,7 +321,7 @@ endif
 
 quiet_cmd_dtc = DTC     $@
 cmd_dtc = $(HOSTCC) -E $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $< ; \
-	$(DTC) -O $(patsubst .%,%,$(suffix $@)) -o $@ -b 0 \
+	$(DTC) -O $(patsubst %o,%,$(patsubst .%,%,$(suffix $@))) -o $@ -b 0 \
 		$(addprefix -i,$(dir $<) $(DTC_INCLUDE)) $(DTC_FLAGS) \
 		-d $(depfile).dtc.tmp $(dtc-tmp) ; \
 	cat $(depfile).pre.tmp $(depfile).dtc.tmp > $(depfile)
@@ -329,6 +329,9 @@ cmd_dtc = $(HOSTCC) -E $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $< ;
 $(obj)/%.dtb: $(src)/%.dts $(DTC) FORCE
 	$(call if_changed_dep,dtc)
 
+$(obj)/%.dtbo: $(src)/%.dtso $(DTC) FORCE
+	$(call if_changed_dep,dtc)
+
 DT_CHECKER ?= dt-validate
 DT_BINDING_DIR := Documentation/devicetree/bindings
 # DT_TMP_SCHEMA may be overridden from Documentation/devicetree/bindings/Makefile
-- 
2.28.0

